src/game_battler.cpp

```diff
@@ -659,3 +665,31 @@ void Game_Battler::Flash(int r, int g, int b, int power, int frames) {
 	flash.time_left = frames;
 }
 
+int Game_Battler::ChangeAtkModifier(int modifier) {
+	const auto prev = atk_modifier;
+	const auto base = GetBaseAtk();
+	SetAtkModifier(Utils::Clamp(atk_modifier + modifier, base / 2, base * 2));
+	return atk_modifier - prev;
+}
+
+int Game_Battler::ChangeDefModifier(int modifier) {
+	const auto prev = def_modifier;
+	const auto base = GetBaseDef();
+	SetDefModifier(Utils::Clamp(def_modifier + modifier, base / 2, base * 2));
+	return def_modifier - prev;
+}
+
+int Game_Battler::ChangeSpiModifier(int modifier) {
+	const auto prev = spi_modifier;
+	const auto base = GetBaseSpi();
+	SetSpiModifier(Utils::Clamp(spi_modifier + modifier, base / 2, base * 2));
+	return spi_modifier - prev;
+}
+
+int Game_Battler::ChangeAgiModifier(int modifier) {
+	const auto prev = agi_modifier;
+	const auto base = GetBaseAgi();
+	SetAgiModifier(Utils::Clamp(agi_modifier + modifier, base / 2, base * 2));
+	return agi_modifier - prev;
+}
+
```

Comment:
I have tested your changes and noticed weird behaviors in relation to stat changes: As soon a stat gets changed (no matter if buff or debuff), the affected stat is then at least 1.5x the base value which makes debuffs no longer work. The clamping makes the modifier at least half of the base value, this explains the 1.5x issue. The modifier must be in the range between `-base / 2` and `+base` because unchanged stats have a modifier of 0 and the calculation of a stat is `base + modifier`.

Reply:
Fixed and added unit tests